Drop-In/src/main/java/com/braintreepayments/api/dropin/adapters/AvailablePaymentMethodNonceList.java

```diff
@@ -0,0 +1,67 @@
+package com.braintreepayments.api.dropin.adapters;
+
+import android.content.Context;
+
+import com.braintreepayments.api.dropin.DropInRequest;
+import com.braintreepayments.api.dropin.utils.PaymentMethodType;
+import com.braintreepayments.api.models.CardNonce;
+import com.braintreepayments.api.models.Configuration;
+import com.braintreepayments.api.models.GooglePaymentCardNonce;
+import com.braintreepayments.api.models.PayPalAccountNonce;
+import com.braintreepayments.api.models.PaymentMethodNonce;
+import com.braintreepayments.api.models.VenmoAccountNonce;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+class AvailablePaymentMethodNonceList {
+
+    final private List<PaymentMethodNonce> items;
+
+    AvailablePaymentMethodNonceList(Context context, Configuration configuration, List<PaymentMethodNonce> paymentMethodNonces, DropInRequest dropInRequest, boolean googlePayEnabled, boolean unionpaySupported) {
+        items = new ArrayList<>();
+
+        for (PaymentMethodNonce paymentMethodNonce: paymentMethodNonces) {
+            boolean shouldAddPaymentMethod = false;
+
+            if (paymentMethodNonce instanceof PayPalAccountNonce && dropInRequest.isPayPalEnabled() && configuration.isPayPalEnabled()) {
+                shouldAddPaymentMethod = true;
+            }
+
+            if (paymentMethodNonce instanceof VenmoAccountNonce && dropInRequest.isVenmoEnabled() && configuration.getPayWithVenmo().isEnabled(context)) {
+                shouldAddPaymentMethod = true;
+            }
+
+            if (paymentMethodNonce instanceof CardNonce && dropInRequest.isCardEnabled()) {
+                Set<String> supportedCardTypes =
+                        new HashSet<>(configuration.getCardConfiguration().getSupportedCardTypes());
+                if (!unionpaySupported) {
+                    supportedCardTypes.remove(PaymentMethodType.UNIONPAY.getCanonicalName());
+                }
+                if (supportedCardTypes.size() > 0) {
+                    shouldAddPaymentMethod = true;
+                }
+            }
+
+            if (paymentMethodNonce instanceof GooglePaymentCardNonce && googlePayEnabled) {
+                if (dropInRequest.isGooglePaymentEnabled()) {
+                    shouldAddPaymentMethod = true;
+                }
+            }
```

Comment:
To make this loop a little more efficient, do we need to check the `paymentMethodNonce` type against every payment method class even if we've already found a match? 

If possible we could use a `switch` statement to check against the specific type of payment method, or just convert this into an `if-else` block so that we can exit early.

Reply:
Yeah agreed, we can do an if-else here to make it more efficient.